apps/server/doc/architecture.md

```diff
@@ -118,6 +118,14 @@ This allows us to include the domain modules in different server deployments, wi
 
 Notice that in the above example, the `PublicService` can be used anywhere within the ApiModule, including in the `ExampleUc`, whereas the `InternalRepo` can not.
 
+### Injectables
+
+Any functions that execute some kind of logic should be part of an injectable. This includes service functions, mappers, helpers, repository functions and the like. This ensures that they can be easily replaced in the future, or mocked away for testing purposes.
```

Comment:
I do not completly agree. 
Any helpers utils/should be synchron and only input > output functions without modifications of the inputs. I have a strong opinion that this code snippets should be simple, but tested in the context where they are used too and not mocked. 
A injection say it is possible that a arc boundary can be come in future but for this methodes i do not see this.

If you use a helper function in a method that you want to test, than any variation of for this helper must be possible to produce with the method parameters or pre setups and they are all cases that must be test for this method on this method.
This also include factorys.